plugins/dfu/dfu-format-srec.c

```diff
@@ -264,14 +264,35 @@ dfu_image_from_srec (DfuImage *image,
 				g_debug ("ignoring data at 0x%x as before start address 0x%x",
 					 (guint) rec_addr32, (guint) start_addr);
 			} else {
+				guint bytecnt = 0;
+				guint32 len_hole = rec_addr32 - addr32_last;
+
+				/* fill any holes */
+				if (addr32_last > 0 && len_hole > 0x100000) {
```

Comment:
Is this an emperically measured number or just arbitrarily high? Would be good to note it in a comment in case this comes up later with another device using DFU.

Reply:
It's a sanity check to avoid DoSing the daemon. Doing a million `g_string_append_c()` takes some time :) I'll add a note.